test/client/nonce.js

```
@@ -0,0 +1,616 @@
+                            throw new Error(`Expected correct userIdToken`);
+                        }
+
+                        return {
+                            data: {
+                                smartWallet: {
+                                    ...wallet
+                                }
+                            }
+                        };
+                    }
+
+                    if (data.query.includes('query GetCheckoutDetails')) {
+                        return {
+                            data: {
+                                checkoutSession: {
+                                    cart: {
+                                        intent:  INTENT.CAPTURE,
+                                        amounts: {
+                                            total: {
+                                                currencyCode: 'USD'
+                                            }
+                                        },
+                                        shippingAddress: {
+                                            isFullAddress: false
+                                        }
+                                    },
+                                    flags: {
+                                        isChangeShippingAddressAllowed: false
+                                    },
+                                    payees: [
+                                        {
+                                            merchantId: 'XYZ12345',
+                                            email:      {
+                                                stringValue: 'xyz-us-b1@paypal.com'
+                                            }
+                                        }
+                                    ]
+                                }
+                            }
+                        };
+                    }
+
+                    if (data.query.includes('mutation ApprovePaymentWithNonce')) {
```

Can we throw an error if `ApprovePaymentWithNonce` is not called? Want to make sure the test is validating this path is hit

---

We have got `.expectCalls` so I am expecting the test case will fail if the mutation is not called. Is that correct assumption?

---

That will only guarantee that GQL is called (which could be for many different queries) -- it will not assert that specifically `ApprovePaymentWithNonce` was hit.

---

Done and updated!